connector/spanmetricsconnector/config.go

```diff
@@ -88,6 +98,10 @@ func (c Config) Validate() error {
 	if err != nil {
 		return err
 	}
+	err = validateEventDimensions(c.Events.Enabled, c.Events.Dimensions)
+	if err != nil {
+		return err
+	}
```

Comment:
nit: We can inline the error handling. Also, I've found annotating errors to be quite useful in tracking down the root error, in the absence of stack traces.

I know `validateDimensions` above doesn't do this; feel free to have a go at fixing that as well.

```suggestion
	if err = validateEventDimensions(c.Events.Enabled, c.Events.Dimensions); err != nil {
		return fmt.Errorf("failed validating event dimensions: %w", err)
	}
```